share/ipld/get.go

```diff
@@ -170,16 +176,19 @@ func GetLeaves(ctx context.Context,
 	wg.Wait()
 }
 
-// GetLeavesByNamespace returns as many leaves from the given root with the given namespace.ID as
-// it can retrieve. If no shares are found, it returns both data and error as nil.
-// A non-nil error means that only partial data is returned, because at least one share retrieval
-// failed The following implementation is based on `GetShares`.
+// GetLeavesByNamespace returns leaves and corresponding proof that could be used to verify leaves
+// inclusion. It returns as many leaves from the given root with the given namespace.ID as it can
+// retrieve. If no shares are found, it returns both data and error as nil. If non-nil proofToFill
+// param passed, it will be filled with data required for inclusion verification. A non-nil error
+// means that only partial data is returned, because at least one share retrieval failed. The
+// following implementation is based on `GetShares`.
 func GetLeavesByNamespace(
 	ctx context.Context,
 	bGetter blockservice.BlockGetter,
 	root cid.Cid,
 	nID namespace.ID,
 	maxShares int,
+	proofToFill *Proof,
```

Comment:
[non-blocking] [optional] I don't like this name 100% but also cannot think of a better one. `proofContainer` maybe? But idk, more of just a comment in case someone else reading this has a better idea

Reply:
How about `optionalProof`?

Reply:
Updated for `proofContainer`